clib/cMap.ml

```diff
@@ -193,6 +201,52 @@ struct
     let fold accu (x, v) = F.add x v accu in
     List.fold_left fold F.empty l
 
+  type 'a sequenced =
+    | End
+    | More of M.t * 'a * 'a F.t * 'a sequenced
+
+  let rec seq_cons m rest =
+    match map_prj m with
+    | MEmpty -> rest
+    | MNode {l; v; d; r; _ } -> seq_cons l (More (v, d, r, rest))
+
+  let rec fold_seq f acc = function
+    | End -> acc
+    | More (k, v, m, r) -> f k v @@ fold_seq f (F.fold f m acc) r
+
+  let move_to_acc (m, acc) = match map_prj m with
+    | MEmpty -> assert false
+    | MNode {l; v; d; r; _ } -> l, More (v, d, r, acc)
+
+  let rec symmetric_cons ((lm, la) as l) ((rm, ra) as r) =
+    if lm = rm then la, ra
```

Comment:
oh there's a sneaky polymorphic equality here

Reply:
Good catch, I spotted it as well, but was waiting for the verdict on other requests to push :-) Fixed.